classes/ObjectModel.php

```diff
@@ -1134,7 +1134,8 @@ public function validateController($htmlentities = true)
             // Checking for fields validity
             // Hack for postcode required for country which does not have postcodes
             if (!empty($value) || $value === '0' || ($field == 'postcode' && $value == '0')) {
-                if (isset($data['validate']) && !Validate::$data['validate']($value) && (!empty($value) || $data['required'])) {
+                $data_validate = $data['validate'];
+                if (isset($data['validate']) && !Validate::$data_validate($value) && (!empty($value) || $data['required'])) {
```

Comment:
It seems PHP notices can be thrown here. You get the property `validate` of $data before checking if it exists. Here is an (untested) code which could fix it.

```php
if (isset($data['validate'])) {
    $data_validate = $data['validate'];
    if (!Validate::$data_validate($value) && (!empty($value) || $data['required'])) {
    [...]
```